Extensions/blacklist.js

```diff
@@ -1174,6 +1178,287 @@ XKit.extensions.blacklist = new Object({
+				XKit.window.show(
+					'No Blacklisted Words',
+					"You don't have any blacklisted words to export!",
+					'error',
+					'<div id="xkit-close-message" class="xkit-button">Close</div>',
+				);
+				return;
+			}
+
+			const selectAll = () =>
+				blacklistItemData.forEach(({ tagCheckbox, contentCheckbox, updateAlreadyFiltered }) => {
+					tagCheckbox.checked = true;
+					contentCheckbox.checked = true;
+					updateAlreadyFiltered();
+				});
+
+			const selectNone = () =>
+				blacklistItemData.forEach(({ tagCheckbox, contentCheckbox, updateAlreadyFiltered }) => {
+					tagCheckbox.checked = false;
+					contentCheckbox.checked = false;
+					updateAlreadyFiltered();
+				});
+
+			const doExport = () => {
+				const newTagWords = blacklistItemData
+					.filter(({ tagCheckbox }) => tagCheckbox.checked)
+					.flatMap(({ textInput }) => textInput.value.split(','))
+					.map(word => word.replace('#', '').trim())
+					.filter(word => !currentFilteredTags.includes(word));
+
+				const newContentWords = blacklistItemData
+					.filter(({ contentCheckbox }) => contentCheckbox.checked)
+					.flatMap(({ textInput }) => textInput.value.split(','))
+					.map(word => word.replace('#', '').trim())
+					.filter(word => !currentFilteredContent.includes(word));
+
+				if (newTagWords.length || newContentWords.length) {
+					Promise.all([
+						newTagWords.length &&
+							apiFetch('/v2/user/filtered_tags', {
+								method: 'POST',
+								body: { filtered_tags: newTagWords },
+							}),
+						newContentWords.length &&
+							apiFetch('/v2/user/filtered_content', {
+								method: 'POST',
+								body: { filtered_content: newContentWords },
+							}),
+					]).then(() => XKit.window.show(
+						'Success',
+						`
+							${newTagWords.length ? `<div>Added filtered tags: ${newTagWords.map(word => `#${word}`).join(', ')}</div>` : ''}
+							${newContentWords.length ? `<div>Added filtered content: ${newContentWords.join(', ')}</div>` : ''}
+							<div>
+								You can see your native filtered content in your
+								<a href="https://www.tumblr.com/settings/account" target="_blank">
+								Tumblr account settings</a>.
+							</div>
+						`,
+						'info',
+						'<div id="xkit-close-message" class="xkit-button">Close</div>',
+					)).catch(showNativeExportError);
+
+				}
+			};
+
+			const createRow = (data) =>
+				dom('tr', null, null, data.map((contents) => dom('td', null, null, [contents])));
+
+			const rows = blacklistItemData.map(({ name, textInput, tagCheckbox, contentCheckbox }) =>
+				createRow([name, textInput, tagCheckbox, contentCheckbox])
+			);
+
+			const containerId = 'blacklist-native-container';
+			const tableId = 'blacklist-native-export-table';
+			const tableBodyId = 'blacklist-native-export-table-body';
+			const doExportId = 'blacklist-native-export-do-export';
+			const selectAllId = 'blacklist-native-export-select-all';
+			const selectNoneId = 'blacklist-native-export-select-none';
```

Comment:
Would you un-extract all of them, or just those not referenced when adding handlers?

Reply:
I would suggest extracting all of them. Although it may seem "less DRY" to have the same string in the code multiple times, repeating an ID (or any other "string-like" constant) is actually just the same as repeating a variable name—and in fact, it's easier, because it's only one name to remember instead of two names that mean the same thing but are different for some reason. 

The only countervailing reason I might see not to extract them is that it's pretty annoying to the reader to have to keep skimming past `blacklist-native-export-` in every. single. id. It adds visual noise that makes it harder to get the the meat of the identifier. So i maybe might suggest using something shorter, like `xkit-bne-container`

Reply:
I hear you. The advantage specifically for ids/classnames that are later referenced when adding handlers, in my opinion, is the inability to misspell the string (since misspelled variable won't pass the linter, but misspelled hardcoded strings will.)

Of course, this is still an issue when matching the ids/classnames to the css file unless one is using a css-in-js solution, so one has to be judicious about these things anyway in this kind of codebase.